pkg/distributor/ha_tracker_test.go

```diff
@@ -792,6 +791,58 @@ func TestCheckReplicaCleanup(t *testing.T) {
 	))
 }
 
+// TestStartingSync tests that values are synced to KVStore on startup
+func TestStartingSync(t *testing.T) {
+	t.Run("check that if a second replica is set up during preLoad that it does not write to it since its not the leader", func(t *testing.T) {
+		cluster := "c1"
+		replica := "r1"
+		var c *haTracker
+		var err error
+		var now time.Time
+
+		codec := GetReplicaDescCodec()
+		kvStore, closer := consul.NewInMemoryClient(codec, log.NewNopLogger(), nil)
+		t.Cleanup(func() { assert.NoError(t, closer.Close()) })
+
+		mock := kv.PrefixClient(kvStore, "prefix")
+		c, err = newHATracker(HATrackerConfig{
+			EnableHATracker:        true,
+			KVStore:                kv.Config{Mock: mock},
+			UpdateTimeout:          15 * time.Second,
+			UpdateTimeoutJitterMax: 0,
+			FailoverTimeout:        time.Millisecond * 2,
+		}, trackerLimits{maxClusters: 100}, nil, log.NewNopLogger())
+		require.NoError(t, err)
+		require.NoError(t, services.StartAndAwaitRunning(context.Background(), c))
+
+		now = time.Now()
+		err = c.checkReplica(context.Background(), "user", cluster, replica, now)
+		assert.NoError(t, err)
+
+		// Check to see if the value in the trackers cache is correct.
+		checkReplicaTimestamp(t, time.Second, c, "user", cluster, replica, now)
+		services.StopAndAwaitTerminated(context.Background(), c) //nolint:errcheck
+
+		replicaTwo := "r2"
+		c, err = newHATracker(HATrackerConfig{
+			EnableHATracker:        true,
+			KVStore:                kv.Config{Mock: mock},
+			UpdateTimeout:          15 * time.Second,
+			UpdateTimeoutJitterMax: 0,
+			FailoverTimeout:        time.Millisecond * 2,
+		}, trackerLimits{maxClusters: 100}, nil, log.NewNopLogger())
+
+		require.NoError(t, err)
+		require.NoError(t, services.StartAndAwaitRunning(context.Background(), c))
+		defer services.StopAndAwaitTerminated(context.Background(), c) //nolint:errcheck
```

Comment:
let's use `t.Cleanup` instead of `defer` like in the consul instance above

Reply:
ill make that change. ty.